runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessConfigureView/ProcessCredentials/ProcessCredentials.tsx

```diff
@@ -0,0 +1,154 @@
+    const actionCredentials = useMemo(
+        () => credentialTypes ? sortByActionCredentialType(processCredentials, credentialTypes) : null,
+        [processCredentials, credentialTypes]
+    );
+
+    const [columns, setColumns] = useState<ActionSortedColumns>([]);
+    const [isAddDialogOpen, setIsAddDialogOpen] = useState(false);
+    const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+    const [currentCredentialId, setCurrentCredentialId] = useState<CredentialId>(null);
+    const [currentActionName, setCurrentActionName] = useState<string | null>(null);
+
+    const { width: windowWidth } = useWindowSize();
+    const rowCount = Math.max(Math.ceil((Math.abs(windowWidth - MARGIN_LIMIT)) / ACTION_MIN_WIDTH), 1);
+
+    const handleDelete = () => {
+        dispatch(processActions.deleteProcessCredential({ resourceId: currentCredentialId }))
+            .unwrap()
+            .then(() => {
+                enqueueSnackbar(
+                    translate('Process.Configure.Credentials.Modal.Delete.Info.Success'),
+                    { variant: 'success' }
+                );
+                dispatch(processActions.getProcessCredentials({ resourceId: String(processId) }));
+            })
+            .catch(() => {
+                enqueueSnackbar(
+                    translate('Process.Configure.Credentials.Modal.Delete.Info.Error'),
+                    { variant: 'error' }
+                );
+            });
+
+        setIsDeleteDialogOpen(false);
+    };
+
+    const handleAddDialogOpen = (actionName: string) => {
+        setCurrentActionName(actionName);
+        setIsAddDialogOpen(true);
+    };
+
+    const handleAddDialogClose = () => {
+        setIsAddDialogOpen(false);
+    };
+
+    const handleDeleteDialogOpen = (credentialId: string) => {
+        setIsDeleteDialogOpen(true);
+        setCurrentCredentialId(credentialId);
+    };
+
+    const handleDeleteDialogClose = () => {
+        setIsDeleteDialogOpen(false);
+    };
+
+    useEffect(() => {
+        if (actionCredentials) {
+            const actionSortedColumns = sortByColumns(actionCredentials, rowCount);
+            setColumns(actionSortedColumns);
+        }
+    }, [windowWidth, actionCredentials]);
+
+    useEffect(() => {
+        dispatch(processActions.getProcessCredentials({ resourceId: String(processId) }));
+    }, []);
+
+    return (
+        <Container>
+            <ProcessCredentialsDeleteDialog
+                isOpen={isDeleteDialogOpen}
+                handleClose={handleDeleteDialogClose}
+                handleDelete={handleDelete}
+            />
+            <ProcessCredentialsAddDialog
+                isOpen={isAddDialogOpen}
+                handleClose={handleAddDialogClose}
+                templateName={currentActionName}
+            />
+            <Header>
+                <Image src={LockIcon} alt='lock icon' style={{ filter: 'brightness(0) saturate(100%)' }}/>
```

Comment:
alts should be translated too

Reply:
it's all right here but because of lower priority it's preferred to use styled components instead of style prop.